Fatras/src/Digitization/Channelizer.cpp

```diff
@@ -0,0 +1,154 @@
+    if (bstart[0] != bend[0]) {
+      double k = segment2d.y() / segment2d.x();
+      double d = start.y() - k * start.x();
+
+      const auto& xboundaries = segmentation.binningData()[0].boundaries();
+      std::vector<double> xbbounds = {
+          xboundaries.begin() + std::min(bstart[0], bend[0]) + 1,
+          xboundaries.begin() + std::max(bstart[0], bend[0]) + 1};
+      for (const auto x : xbbounds) {
+        cSteps.push_back(ChannelStep{
+            {(bstart[0] < bend[0] ? 1 : -1), 0}, {x, k * x + d}, start});
+      }
+    }
+    // The lines channel segment lines along y
+    if (bstart[1] != bend[1]) {
+      double k = segment2d.x() / segment2d.y();
+      double d = start.x() - k * start.y();
+      const auto& yboundaries = segmentation.binningData()[1].boundaries();
+      std::vector<double> ybbounds = {
+          yboundaries.begin() + std::min(bstart[1], bend[1]) + 1,
+          yboundaries.begin() + std::max(bstart[1], bend[1]) + 1};
+      for (const auto y : ybbounds) {
+        cSteps.push_back(ChannelStep{
+            {0, (bstart[1] < bend[1] ? 1 : -1)}, {k * y + d, y}, start});
+      }
+    }
+
+  } else if (surface.type() == Acts::Surface::SurfaceType::Disc) {
+    Acts::Vector2D pstart(Acts::VectorHelpers::perp(start),
+                          Acts::VectorHelpers::phi(start));
+    Acts::Vector2D pend(Acts::VectorHelpers::perp(end),
+                        Acts::VectorHelpers::phi(end));
+
+    // Get the segmentation and convert it to lines & arcs
+    bstart = {static_cast<unsigned int>(segmentation.bin(pstart, 0)),
+              static_cast<unsigned int>(segmentation.bin(pstart, 1))};
+    bend = {static_cast<unsigned int>(segmentation.bin(pend, 0)),
+            static_cast<unsigned int>(segmentation.bin(pend, 1))};
+
+    // Fast single channel exit
+    if (bstart == bend) {
+      return {ChannelSegment(bstart, {start, end}, segment2d.norm())};
+    }
+
+    double phistart = pstart[1];
+    double phiend = pend[1];
+
+    // The radial boundaries
+    if (bstart[0] != bend[0]) {
+      const auto& rboundaries = segmentation.binningData()[0].boundaries();
+      std::vector<double> rbbounds = {
+          rboundaries.begin() + std::min(bstart[0], bend[0]) + 1,
+          rboundaries.begin() + std::max(bstart[0], bend[0]) + 1};
+      for (const auto& r : rbbounds) {
+        auto radIntersection =
+            Acts::detail::IntersectionHelper2D::intersectCircleSegment(
+                r, std::min(phistart, phiend), std::max(phistart, phiend),
+                start, (end - start).normalized());
+        cSteps.push_back(ChannelStep{{(bstart[0] < bend[0] ? 1 : -1), 0},
+                                     radIntersection.position,
+                                     start});
+      }
+    }
+    // The phi boundaries
+    if (bstart[1] != bend[1]) {
+      double referenceR = surface.binningPositionValue(geoCtx, Acts::binR);
+      Acts::Vector2D origin = {0., 0.};
```

Comment:
You're right, but it doesn't matter: phi lines need to come from (0,0) and since the segment is required to be inside the surface, the solution is independent form the phi boundary origin. Here, it just saves to calculate an origin for every phi boundary, when they all go through (0,0).

Reply:
I believe you are right.